src/algebra/big_operators/nat_antidiagonal.lean

```diff
@@ -62,6 +62,17 @@ begin
   refl,
 end
 
+@[to_additive]
+lemma prod_antidiagonal {M : Type*} [comm_monoid M]
```

Comment:
```suggestion
lemma prod_antidiagonal_eq_prod_range_succ_mk {M : Type*} [comm_monoid M]
```

Since this is just the lemma above, but with `prod.mk` on the LHS.

If you swap the order of the lemmas, you can golf the proofs to:
```
@[to_additive]
lemma prod_antidiagonal_eq_prod_range_succ_mk {M : Type*} [comm_monoid M] (f : ℕ × ℕ → M) (n : ℕ) :
  ∏ ij in finset.nat.antidiagonal n, f ij = ∏ k in range n.succ, f (k, n - k) :=
begin
  convert prod_map _ ⟨λ i, (i, n - i), λ x y h, (prod.mk.inj h).1⟩ _,
  refl,
end

/-- This lemma unifies better than `finset.nat.prod_antidiagonal_eq_prod_range_succ_mk` when using
`rw ←` -/
@[to_additive]
lemma prod_antidiagonal_eq_prod_range_succ {M : Type*} [comm_monoid M] (f : ℕ → ℕ → M) (n : ℕ) :
  ∏ ij in finset.nat.antidiagonal n, f ij.1 ij.2 = ∏ k in range n.succ, f k (n - k) :=
prod_antidiagonal_eq_prod_range_succ_mk _ _
```

Reply:
Done, thank you!